packages/react-fresh/src/ReactFreshBabelPlugin.js

```diff
@@ -158,6 +163,74 @@ export default function(babel) {
     return false;
   }
 
+  let hookCalls = new WeakMap();
+
+  function recordHookCall(functionNode, hookCallPath, hookName) {
+    if (!hookCalls.has(functionNode)) {
+      hookCalls.set(functionNode, []);
+    }
+    let hookCallsForFn = hookCalls.get(functionNode);
+    let key = '';
+    if (hookCallPath.parent.type === 'VariableDeclarator') {
+      // TODO: if there is no LHS, consider some other heuristic.
+      key = hookCallPath.parentPath.get('id').getSource();
+    }
+    hookCallsForFn.push({
+      name: hookName,
+      callee: hookCallPath.node.callee,
+      key,
+    });
+  }
+
+  function isBuiltinHook(hookName) {
+    switch (hookName) {
+      case 'useState':
+      case 'React.useState':
+      case 'useReducer':
+      case 'React.useReducer':
+      case 'useEffect':
+      case 'React.useEffect':
+      case 'useLayoutEffect':
+      case 'React.useLayoutEffect':
+      case 'useMemo':
+      case 'React.useMemo':
+      case 'useCallback':
+      case 'React.useCallback':
+      case 'useImperativeMethods':
+      case 'React.useImperativeMethods':
```

Comment:
Why the decision to omit `useRef`, `useContext`, and `useDebugValue`?

Reply:
I forgot about them lol

Reply:
😂